### PR TITLE
added file input and output operations

### PR DESCRIPTION
**Pull request by NikitaSutulov**
Added functionality for saving and reading the chart file in specific format.

### CODE REVIEW

File: app/src/main/java/com/nickytoolchick/agraph/fileio/FileReader.kt [hunk 1]
```diff
@@ -1,7 +1,34 @@
...
+import kotlinx.serialization.json.Json
+import java.io.File
+
 class FileReader {
-    // TODO("Not implemented yet")
+    fun readFile(ctx: Context): Pair<ChartOptions, DatasetOptions>? {
+        val file = File(ctx.filesDir, Constants.FILE_NAME)
+
+        if (!file.exists()) {
+            return null
```
Comment (michigang1) [inline app/src/main/java/com/nickytoolchick/agraph/fileio/FileReader.kt]: Dont't return nulls. You can return anything but nulls. It's not safe.
U can use empty data classes, Nothing classes and etc

File: app/src/main/java/com/nickytoolchick/agraph/fileio/FileReader.kt [hunk 2]
```diff
@@ -1,7 +1,34 @@
...
+        }
+
+        try {
+            val lines = file.readLines()
+
+            if (lines.size != 3) {
+                return null
```
Comment (michigang1) [inline app/src/main/java/com/nickytoolchick/agraph/fileio/FileReader.kt]: Dont't return nulls. You can return anything but nulls. It's not safe.
U can use empty data classes, Nothing classes and etc

File: app/src/main/java/com/nickytoolchick/agraph/fileio/FileReader.kt [hunk 3]
```diff
@@ -1,7 +1,34 @@
...
+            }
+
+            val chartOptions = Json.decodeFromString<ChartOptions>(lines[0])
+            val datasetOptions = Json.decodeFromString<DatasetOptions>(lines[2])
+
+            return Pair(chartOptions, datasetOptions)
+        } catch (e: Exception) {
+            return null
```
Comment (michigang1) [inline app/src/main/java/com/nickytoolchick/agraph/fileio/FileReader.kt]: It's better to check exceptions and throw it

**Review by michigang1 — APPROVED** [review]

**Comment by NikitaSutulov** [comment]
Removed nulls from the issued code, replaced with default options objects creation and returning.

**Pull request merged**